tests/unit/systems/fil/test_forest.py

```diff
@@ -0,0 +1,145 @@
+    output_schema = Schema([ColumnSchema("output__0", dtype=np.float32)])
+    config = Forest(model, input_schema).export(tmpdir, input_schema, output_schema, node_id=2)
+    node_config = json.loads(config.parameters[config.name].string_value)
+
+    assert json.loads(node_config["output_dict"]) == {
+        "output__0": {"dtype": "float32", "is_list": False, "is_ragged": False}
+    }
+
+    cls = Forest.from_config(node_config)
+    assert cls.fil_model_name == "2_fil"
+
+
+def read_config(config_path):
+    with open(config_path, "rb") as f:
+        config = model_config.ModelConfig()
+        raw_config = f.read()
+        return text_format.Parse(raw_config, config)
+
+
+def test_export(tmpdir):
+    rows = 200
+    num_features = 16
+    X, y = sklearn.datasets.make_regression(
+        n_samples=rows,
+        n_features=num_features,
+        n_informative=num_features // 3,
+        random_state=0,
+    )
+    model = xgboost.XGBRegressor()
+    model.fit(X, y)
+    feature_names = [str(i) for i in range(num_features)]
+    input_schema = Schema([ColumnSchema(col, dtype=np.float32) for col in feature_names])
+    output_schema = Schema([ColumnSchema("output__0", dtype=np.float32)])
+    _ = Forest(model, input_schema).export(tmpdir, input_schema, output_schema, node_id=2)
+
+    config_path = tmpdir / "2_forest" / "config.pbtxt"
+    parsed_config = read_config(config_path)
+    assert parsed_config.name == "2_forest"
+    assert parsed_config.backend == "python"
+
+    config_path = tmpdir / "2_fil" / "config.pbtxt"
+    parsed_config = read_config(config_path)
+    assert parsed_config.name == "2_fil"
+    assert parsed_config.backend == "fil"
+
+
+def test_ensemble(tmpdir):
+    rows = 200
+    num_features = 16
+    X, y = sklearn.datasets.make_regression(
+        n_samples=rows,
+        n_features=num_features,
+        n_informative=num_features // 3,
+        random_state=0,
+    )
+    feature_names = [str(i) for i in range(num_features)]
+    df = pd.DataFrame(X, columns=feature_names)
+    dataset = Dataset(df)
+
+    # Fit GBDT Model
+    model = xgboost.XGBRegressor()
+    model.fit(X, y)
+
+    input_schema = Schema([ColumnSchema(col, dtype=np.float32) for col in feature_names])
+    selector = ColumnSelector(feature_names)
+
+    workflow_ops = ["0", "1", "2"] >> wf_ops.LogOp()
+    workflow = Workflow(workflow_ops)
+    workflow.fit(dataset)
+
+    triton_chain = selector >> TransformWorkflow(workflow) >> Forest(model, input_schema)
+
+    triton_ens = Ensemble(triton_chain, input_schema)
+
+    triton_ens.export(tmpdir)
+
+    config_path = tmpdir / "1_forest" / "config.pbtxt"
+    parsed_config = read_config(config_path)
+    assert parsed_config.name == "0_forest"
+    assert parsed_config.backend == "python"
+
+    config_path = tmpdir / "1_fil" / "config.pbtxt"
+    parsed_config = read_config(config_path)
+    assert parsed_config.name == "0_fil"
```

Comment:
This should maybe also be `1_fil`, but I'm honestly not 100% sure how (or if) we should account for this op in the ensemble operator id numbering. I think it could be problematic to have two operators with the same id integer, but I don't think this code actually affects the mapping of ids to operators, so maybe aligning the prefix integer with the id of the operator it belongs to makes sense here? Assuming this doesn't break anything, I have no objections to this convention, I'm just thinking out loud here.

Reply:
Yes, I think having the fIl model prefix id match the PredictForest id as it currently does (aside from this test typo) makes sense because they’re part of the same operator conceptually, from the perspective of the user of the library. And it should avoid collisions if there are more than one in the ensemble.